libp2p/upgrademngrs/muxedupgrade.nim

```diff
@@ -121,7 +123,10 @@ method upgradeIncoming*(u: MuxedUpgrade, incomingConn: Connection) {.async, gcsa
     except CatchableError as exc:
       debug "Exception in secure handler during incoming upgrade", msg = exc.msg, conn
       if not cconn.isUpgraded:
-        cconn.upgrade(exc)
+        # signal failure
+        cconn.upgrade(failed = exc)
+      if exc of CancelledError:
```

Comment:
We can't reraise the cancellation here, because is a spawned proc, there is nothing to propagate it to... With the `raises: [Defect]` this would have been obvious.